Mathlib/Algebra/Order/Group/Basic.lean

```diff
@@ -137,3 +137,77 @@ lemma zpow_eq_zpow_iff' (hn : n ≠ 0) : a ^ n = b ^ n ↔ a = b := zpow_left_in
 #align zsmul_eq_zsmul_iff' zsmul_eq_zsmul_iff'
 
 end LinearOrderedCommGroup
+
+section LinearOrderedAddCommGroupWithTop
+
+variable {α : Type*} [LinearOrderedAddCommGroupWithTop α]
+
+@[simp]
+lemma top_ne_zero' :
+    (⊤ : α) ≠ 0 := by
+  intro nh
+  have ⟨a, b, h⟩ := Nontrivial.exists_pair_ne (α := α)
+  have : a + 0 ≠ b + 0 := by simpa
+  rw [← nh] at this
+  simp at this
+
+@[simp]
+lemma neg_eq_top {a : α} :
+    -a = ⊤ ↔ a = ⊤ :=
+  ⟨by
+  intro h
+  by_contra nh
+  replace nh := add_neg_cancel_of_ne_top nh
+  rw [h, add_top] at nh
+  exact top_ne_zero' nh, fun h ↦ h ▸ neg_top⟩
```

Comment:
Since `Iff` is a structure, you can use `where` notation
```suggestion
@[simp] lemma neg_eq_top {a : α} : -a = ⊤ ↔ a = ⊤ where
  mp h := by
    by_contra nh
    replace nh := add_neg_cancel_of_ne_top nh
    rw [h, add_top] at nh
    exact top_ne_zero' nh
  mpr h := h ▸ neg_top
```